src/components/form/date-input/date-input-component.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { By } from '@angular/platform-browser';
+import { DateInputComponent } from './date-input.component';
+import { DebugElement } from '@angular/core';
+import { PaletteUtilsModule } from '../../../shared/components/palette/utils/utils.module';
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import { MockComponent } from 'ng2-mock-component';
+import createSpy = jasmine.createSpy;
+
+describe('Date input component', function () {
+
+    let fixture: ComponentFixture<DateInputComponent>;
+    let component: DateInputComponent;
+    let de: DebugElement;
+
+    let Input: any;
+    let onChange;
+
+    beforeEach(async(() => {
+        // Input is mocked so that one-way bound inputs can be tested
+        Input = MockComponent({
+            selector: 'input', inputs: [
+                'type',
+                'value',
+                'change',
+                'keyup',
+                'disabled'
+            ]
+        });
+        TestBed
+            .configureTestingModule({
+                imports: [
+                    ReactiveFormsModule,
+                    PaletteUtilsModule
+                ],
+                declarations: [
+                    DateInputComponent,
+
+                    // Mock
+                    Input
+                ],
+                providers: []
+            })
+            .compileComponents();
+
+        fixture = TestBed.createComponent(DateInputComponent);
+        component = fixture.componentInstance;
+        onChange = createSpy('onChange');
+
+        component.registerOnChange(onChange);
+
+        de = fixture.debugElement;
+        fixture.detectChanges();
+    }));
+
+    it('should valid when date value', async () => {
+        component.id = 'dateField'
+        component.writeValue('2021-04-09T08:02:27.542');
+        fixture.detectChanges();
+        let monthInput = await de.query(By.css(`#${component.monthId()}`)).componentInstance;
+        expect(monthInput.value).toBe('04');
+        let dayInput = await de.query(By.css(`#${component.dayId()}`)).componentInstance;
+        expect(dayInput.value).toBe('09');
+        let yearInput = await de.query(By.css(`#${component.yearId()}`)).componentInstance;
+        expect(yearInput.value).toBe('2021');
+    });
+    it('should be valid when date', () => {
+        let results = component.validate({ value: '2018-04-09T08:02:27.542' } as FormControl);
+        expect(results).toBeUndefined();
+    });
+
+    it('should be invalid when invalid date pattern', () => {
+        let results = component.validate({ value: 'x' } as FormControl);
+        expect(results['pattern']).toBeTruthy();
+    });
+
+    it('should not be valid when empty string and mandatory', () => {
+        component.mandatory = true;
+        let results = component.validate({ value: '' } as FormControl);
+        expect(results).toEqual({ required: 'This field is required' });
```

Comment:
Is there a string constant you can use from a validator somewhere? You're testing for a hard-coded string value here and in a test below. I'd have thought the `validate(...)` method on the component would be using a string constant rather than a hard-coded string... and if it's not, it should.